zebra-consensus/src/error.rs

```diff
@@ -146,4 +146,7 @@ pub enum BlockError {
         zebra_chain::work::difficulty::ExpandedDifficulty,
         zebra_chain::parameters::Network,
     ),
+
+    #[error("wrong network upgrade")]
+    WrongNetworkUpgrade,
```

Comment:
The serialized field is a consensus branch id, so our errors should focus on the error in the data:
```suggestion
    #[error("wrong consensus branch id for network upgrade")]
    WrongConsensusBranchId,
```